python/Ganga/Core/GangaRepository/Registry.py

```diff
@@ -225,81 +236,35 @@ def unlock_transaction(self, this_id):
         self.hard_lock[this_id].release()
 
     # Methods intended to be called from ''outside code''
+    @synchronised
     def __getitem__(self, this_id):
         """ Returns the Ganga Object with the given id.
             Raise RegistryKeyError"""
         logger.debug("__getitem__")
-        #self._lock.acquire()
         try:
-            self.lock_transaction( this_id, "_getitem")
-
-            real_id = None
-            if type(this_id) is int:
-                if this_id >= 0:
-                    ## +ve integer, should be in dictionary
-                    real_id = this_id
-                    this_obj = self._objects[this_id]
-                else:
-                    ## -ve integer should be a relative object in dictionary
-                    real_id = self._objects.keys()[this_id]
-                    this_obj = self._objects[real_id]
-            else:
-                ## NOT an integer, maybe it's a slice or other?
-                this_obj = self._objects[this_id]
-
-            logger.debug("found_object")
-
-            found_id = None
-            if hasattr(this_obj, _id_str):
-                found_id = getattr(this_obj, _id_str)
-            if hasattr(this_obj, _reg_id_str):
-                found_id = getattr(this_obj, _reg_id_str)
-            if found_id is not None and real_id is not None:
-                assert( found_id == real_id )
```

Comment:
So, a possible example would be:

* Thread 1 does something to an object that locks that object but later (before releasing the object lock) requires a lock on the registry
* Thread 2 then requests the same object from the registry before the other thread has a lock on the registry, thus locking it first.
* Thread 1 is then waiting to get the registry lock from Thread 2
* But, Thread 2 (because of the request for the registry ID from the object) is waiting on for Thread 1 to release it's lock on the object.
* and you're stuck...

Now, I'm aware I haven't listed specifics but seeing as this **can** happen (and Matt was seeing it), for the sake of some code that I would prefer to be in a test case anyway, it's better to simplify everything as much as possible.